### PR TITLE
[CAMEL-14871] doneFileName attribute on the S3Consumers

### PR DESCRIPTION
**Pull request by vilmosnagy**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by oscerd — CHANGES_REQUESTED** [review]
I would prefer this to be done on aws2-s3component. Aws s3 will be deprecated and probably removed in next lts release.

**Comment by vilmosnagy** [comment]
@oscerd I've implemented it on both of them. :slightly_smiling_face:

**Comment by oscerd** [comment]
Ah great! I was from mobile and didn't spot it. I'll review today :-) thanks a lot!

**Comment by vilmosnagy** [comment]
@oscerd AFAIK this is my first meaningful contribution for Camel - we've run this code in a test environment of ours for a while now, but I'm not sure that I've followed all of the documentation guides and other stuff correctly, so every feedback is welcomed.

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java [hunk 1]
```diff
@@ -16,19 +16,11 @@
  */
 package org.apache.camel.component.aws.s3;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Queue;
+import java.util.*;
```
Comment (oscerd) [inline components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java]: Please avoid * imports

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java [hunk 2]
```diff
@@ -16,19 +16,11 @@
...
 
 import com.amazonaws.AmazonClientException;
 import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.model.GetObjectRequest;
-import com.amazonaws.services.s3.model.ListObjectsRequest;
-import com.amazonaws.services.s3.model.ObjectListing;
-import com.amazonaws.services.s3.model.S3Object;
-import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.*;
```
Comment (oscerd) [inline components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java]: Same

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java [hunk 3]
```diff
@@ -66,9 +67,12 @@ protected int poll() throws Exception {
 
         String fileName = getConfiguration().getFileName();
         String bucketName = getConfiguration().getBucketName();
+        String doneFileName = getConfiguration().getDoneFileName();
         Queue<Exchange> exchanges;
 
-        if (fileName != null) {
+        if (shouldSkipCauseDoneFileIsConfiguredButMissing(bucketName, doneFileName)) {
```
Comment (oscerd) [inline components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java:73]: Naming is hard but this name is really long :-D

**Review by oscerd — APPROVED** [review]
Just a minor finding about imports. Seems to be ok.

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java (hunk 2 above)
Comment (vilmosnagy) [reply components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java]: Yeah, fixed, thanks.

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java (hunk 1 above)
Comment (vilmosnagy) [reply components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java]: Yeah, fixed, thanks.

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java (hunk 3 above)
Comment (vilmosnagy) [reply components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java:73]: What do you think about the following:

```java
if (!doneFileCheckPasses(bucketName, doneFileName)) {
```

Or I was thinking about something like:

```java
if (doneFileCheckPasses(bucketName, doneFileName)) {
    // keep everything that was here beforehand
} else {
    exchanges = new LinkedList<>();
}
```


but that would mean a lot of whitespace changes and I don't want to create unnecessary changes.

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
I'm going to merge this today

**Pull request merged**